### PR TITLE
feat: add repo stub and tests

### PR DESCRIPTION
**Pull request by nazzrrg**
+ refactor

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+	"gopkg.in/yaml.v3"
 )
 
 type Config struct {
-	ForwarderConfig  *forwarder.Config   `yaml:"forwarder_config" validate:"required"`
-	HttpServerConfig *http_server.Config `yaml:"http_config" validate:"required"`
+	ForwarderConfig  *forwarder.Config       `yaml:"forwarder_config" validate:"required"`
+	HttpServerConfig *http_server.Config     `yaml:"http_config" validate:"required"`
+	TempConfig       *TempConfig             `yaml:"temp_config" validate:"required"`
+	MetricsConfig    *metrics_handler.Config `yaml:"temp_config"`
```
Comment (jvstme) [inline internal/config/config.go:18]:
```
panic: duplicated key 'temp_config' in struct config.Config [recovered]
```
Also consider making required. I'm not sure what will happen if this is not specified, but if that will mean no auth for `/metrics` then that's a security issue

File: internal/services/forwarder/service.go [hunk 2]
```diff
@@ -18,51 +20,62 @@ var messageTypeIcons = map[entities.MessageType]string{
...
 
 	for {
 		select {
-		case message := <-f.q.AsChan():
+		case <-ctx.Done():
+			s.l.Info("stopping forwarder service")
+			return nil
+		case message := <-s.q.AsChan():
+			l := s.l.With("interactionId", message.InteractionId, "fromVkSenderId", message.VkSenderId)
+			interaction, err := s.repo.GetInteraction(message.InteractionId)
```
Comment (jvstme) [inline internal/services/forwarder/service.go:55]: (nit) I'm a bit concerned that this is the second repo request while handling the same VK message. First we request the Interaction in the VK webhook handler to check that it exists, and then we request the same Interaction here.

Maybe consider extracting `TgChatId` from the Interaction in the handler and putting it to the queue. I.e. replacing `InteractionId` with `TgChatId` in the `Message` struct

File: internal/services/forwarder/service_test.go [hunk 3]
```diff
@@ -59,3 +69,155 @@ func TestRender(t *testing.T) {
...
+		})
+
+		cancel()
+		wg.Wait()
+
+		logOutput := buf.String()
+		assert.Contains(t, logOutput, `get interaction failed`)
+		assert.Contains(t, logOutput, `interactionId=""`)
+	})
+	t.Run("empty interaction", func(t *testing.T) {
```
Comment (jvstme) [inline internal/services/forwarder/service_test.go:112]: Wasn't the previous test about empty interaction? Maybe the test names in this test and the previous test are swapped by mistake

File: internal/services/http_server/handlers/vk_callback_handler/handler.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+	text := fmt.Sprintf("unsupported message type: %s", dto.Type)
+	h.l.Warn(text, "messageType", dto.Type)
+	ctx.Error(text, fasthttp.StatusBadRequest)
+}
+
+func (h *vkCallbackHandler) handleChallenge(ctx *fasthttp.RequestCtx) error {
+	hookId, ok := ctx.UserValue(HookIdKey).(string)
+	if !ok || hookId == "" {
+		// should be impossible by still check
```
Comment (jvstme) [inline internal/services/http_server/handlers/vk_callback_handler/handler.go:78]:
```suggestion
		// should be impossible but still check
```

File: internal/services/http_server/handlers/vk_callback_handler/handler.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	ctx.Response.SetStatusCode(fasthttp.StatusOK)
+	ctx.Response.Header.SetContentType("text/plain")
+	ctx.Response.SetBody([]byte(interaction.ConfirmationString))
+	return nil
+}
+
+func (h *vkCallbackHandler) handleMessage(ctx *fasthttp.RequestCtx, messageType entities.MessageType) error {
+	hookId, ok := ctx.UserValue(HookIdKey).(string)
+	if !ok || hookId == "" {
+		// should be impossible by still check
```
Comment (jvstme) [inline internal/services/http_server/handlers/vk_callback_handler/handler.go:99]:
```suggestion
		// should be impossible but still check
```

File: internal/services/http_server/service_test.go [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+				confirmationString: "confirmationString",
+				reqBody:            []byte(`{"type":"confirmation","event":"test_event_id","v":"1.0","group_id":12345}`),
+				url:                "http://localhost:8080/api/vk/callback/hookId",
+			},
+			statusCode:   200,
+			bodyContains: []string{"confirmationString"},
+		},
+		// handleMessage:
+		{
+			name: "handleChallenge[message_new]: no hook id",
```
Comment (jvstme) [inline internal/services/http_server/service_test.go:123]: Here and in all tests below
```suggestion
			name: "handleMessage[message_new]: no hook id",
```

**Review by jvstme — COMMENTED** [review]
Looks very good overall, thanks! I had issues running this though, so some tweaks might be needed. Please see my comments below

File: internal/services/forwarder/service_test.go (hunk 3 above)
Comment (nazzrrg) [reply internal/services/forwarder/service_test.go:112]: until the implementation is not clear I consider nil err and nil interaction a valid case

File: internal/services/forwarder/service.go (hunk 2 above)
Comment (nazzrrg) [reply internal/services/forwarder/service.go:55]: No performance issues in foreseeable future, so let it be, as it seems more flexible this way (e.g. may come the need to extract more information in forwarder service)

**Pull request merged**